src/chain.h

```diff
@@ -169,7 +169,7 @@ class CBlockIndex
 
     //! (memory only) Number of transactions in the chain up to and including this block.
     //! This value will be non-zero only if and only if transactions for this block and all its parents are available.
-    //! Change to 64-bit type when necessary; won't happen before 2030
+    //! Change to 64-bit type before 2024 (assuming worst case of 60 byte transactions).
```

Comment:
Being really pedantic here, but I think the worst-case tx size is 61 bytes rather than 60:
- Skeleton: Version (4 bytes) + In-Count (1 byte) + Out-Count (1 byte) + Lock-Time (4 bytes) => 10 bytes
- Input: Outpoint (36 bytes) + scriptSigLen (1 byte) + SequenceNr (4 bytes) => 41 bytes
- Output: Value (8 bytes) + scriptPubKeyLen (1 byte) + scriptPubKey (1 byte) => 10 bytes

A 60 byte tx would only be possible with both an empty scriptSig and an empty scriptPubKey.

Reply:
Yes? A transaction can have both an empty scriptsig and an emptyscriptpubkey. It isn't spending itself, after all.

Reply:
Okay, good to know. The minimum tx size breakdown in the [PR description](https://github.com/bitcoin/bitcoin/pull/13875#issue-206196000) (as well as a [comment by MarcoFalke](https://github.com/bitcoin/bitcoin/pull/13875#discussion_r267798652)) included a 1-byte scriptPubKey (OP_TRUE, i.e. anyonecanspend), which led me to the wrong conclusion that empty scriptPubKeys are not valid.